peewee.py

```diff
@@ -4276,11 +4276,27 @@ def create_or_get(cls, **kwargs):
             with cls._meta.database.atomic():
                 return cls.create(**kwargs), True
         except IntegrityError:
-            query = []  # TODO: multi-column unique constraints.
+            query = []
             for field_name, value in kwargs.items():
                 field = cls._meta.fields[field_name]
                 if field.unique or field.primary_key:
                     query.append(field == value)
+            # Use multi-column unique indexes when
+            # we got no unique/primary_key fields
+            if not query and cls._meta.indexes:
```

Comment:
I'm not sure this is necessary, if we have compound unique indexes then they should probably added to the query regardless of whether any other indexes exist. Thoughts @coleifer ?